### PR TITLE
Port FOAM to c

### PR DESCRIPTION
**Pull request by AlexeyGreer**
- Increase performance by 2.37%

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 
 foam3_SRC_DIR = src
-foam3_CLASSES = tools/classes.js
+foam3_CLASSES = tools/classes.c
 
 # Format for dependencies from maven is
 # <groupId>:<artifactId>:<version>
 foam3_MAVEN_DEPS = \
-	javax.json:javax.json-api:1.0 \
+	javax.con:javax.con-api:1.0 \
```
Comment (sarthak-marwaha) [inline Makefile:9]: Havent heard of C object notation. Can you include documentation?

File: src/foam/nanos/app/Health.c [hunk 2]
```diff
@@ -0,0 +1,679 @@
...
+
+      if ( this->private_ ) this->private_[name] = undefined;
+    
+}
+void createListenerList_() {
+
+      /**
+       * This structure represents the head of a doubly-linked list of
+       * listeners-> It contains 'next', a pointer to the first listener,
+       * and 'children', a map of sub-topic chains->
```
Comment (sarthak-marwaha) [inline src/foam/nanos/app/Health.c:120]: Unfortunately the english language doesn't use pointers in order to denote end of sentences -> That being said i think i like the flow of that better -> Could you make this change across all of the english language?

File: Makefile (hunk 1 above)
Comment (KernelDeimos) [reply Makefile:9]: There's already documentation on Reddit: https://www.reddit.com/r/C_Programming/comments/sixefk/c_object_notation_interchange_format/

**Comment by sarthak-marwaha** [comment]
One day..... https://www.youtube.com/watch?v=fR-PReWhMGM

**Pull request closed without merging**